include/arch/arm/cortex_m/mpu/arm_mpu_v8m.h

```diff
@@ -165,7 +215,44 @@ struct arm_mpu_region_attr {
 		__is_writable__; \
 	})
 
+/*
+ * @brief Evaluate Execution allowance
+ *
+ * Evaluate whether the access permissions include execution.
+ *
+ * @param attr The k_mem_partition_attr_t object holding the
+ *             MPU attributes to be checked against execution
+ *             allowance.
+ */
 #define K_MEM_PARTITION_IS_EXECUTABLE(attr) \
-	(!((attr) & (NOT_EXEC)))
+	(!((attr.rbar) & (NOT_EXEC)))
+
+/* Attributes for no-cache enabling (share-ability is selected by default) */
+
+/* Read-Write access permission attributes */
+#define K_MEM_PARTITION_P_RW_U_RW_NOCACHE \
+	{(P_RW_U_RW_Msk | NOT_EXEC | OUTER_SHAREABLE_Msk), \
+		MPU_MAIR_INDEX_SRAM_NOCACHE}
+#define K_MEM_PARTITION_P_RW_U_NA_NOCACHE \
+	{(P_RW_U_NA_Msk | NOT_EXEC | OUTER_SHAREABLE_Msk), \
+		MPU_MAIR_INDEX_SRAM_NOCACHE}
+#define K_MEM_PARTITION_P_RO_U_RO_NOCACHE \
+	{(P_RO_U_RO_Msk | NOT_EXEC | OUTER_SHAREABLE_Msk), \
+		MPU_MAIR_INDEX_SRAM_NOCACHE}
+#define K_MEM_PARTITION_P_RO_U_NA_NOCACHE \
+	{(P_RO_U_NA_Msk | NOT_EXEC | OUTER_SHAREABLE_Msk), \
+		MPU_MAIR_INDEX_SRAM_NOCACHE}
+
+/* Execution-allowed attributes */
+#define K_MEM_PARTITION_P_RWX_U_RWX_NOCACHE \
+	{(P_RW_U_RW_Msk | OUTER_SHAREABLE_Msk), MPU_MAIR_INDEX_SRAM_NOCACHE}
+#define K_MEM_PARTITION_P_RX_U_RX_NOCACHE \
+	{(P_RO_U_RO_Msk | OUTER_SHAREABLE_Msk), MPU_MAIR_INDEX_SRAM_NOCACHE}
+
 #endif /* _ASMLANGUAGE */
-#endif /* USERSPACE */
+
+#define _ARCH_MEM_PARTITION_ALIGN_CHECK(start, size) \
```

Comment:
Is this check valid for v8 MPU?

Reply:
Ah, good point! I already updated the macro for ARMv8-M to remove the requirement for alignment with size.
But, of course, I should have removed, also the requirement for power-of-two!